File: pinkmoe_admin/src/api/content/post.ts
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-05-08 01:11:49
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 09:14:07
 * @FilePath: /pinkmoe_admin/src/api/content/post.ts
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe   (如需用于商业用途或者二开，请联系作者捐助任意金额即可)
 * QQ:2419857357;支付宝:13135986153
 * Copyright (c) 2022 by coderzhaolu, All Rights Reserved. 
 */
import { http } from '@/utils/http/axios';
import { BasicResponseModel } from '@/api/common';

/**
 * @description: 获取文章列表
 * @param params
 */
export function getPostList(params?) {
  return http.request<BasicResponseModel>(
    {
      url: '/Admin/Post/PostList',
      method: 'GET',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 上传文件
 * @param params
 */
export function upload(params?) {
  return http.request<BasicResponseModel>(
    {
      url: '/Admin/Upload/FileUpload',
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 更新文章
 * @param params
 */
export function updatePost(params) {
  return http.request(
    {
      url: `/Admin/Post/PostUpdate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 更新文章
 * @param params
 */
export function updatePostStatus(params) {
  return http.request(
    {
      url: `/Admin/Post/PostUpdateStatus`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 创建文章
 * @param params
 */
export function createPost(params) {
  return http.request(
    {
      url: `/Admin/Post/PostCreate`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

/**
 * @description: 删除文章
 * @param params
 */
export function deletePost(params) {
  return http.request(
    {
      url: `/Admin/Post/PostDelete`,
      method: 'POST',
      params,
    },
    {
      isTransformResponse: false,
    }
  );
}

File: pinkmoe_server/controller/admin/post.go
/*
 * @Author: coderzhaolu && izhaicy@163.com
 * @Date: 2022-06-22 11:13:07
 * @LastEditors: coderzhaolu && izhaicy@163.com
 * @LastEditTime: 2022-08-07 09:02:01
 * @FilePath: /pinkmoe_server/controller/admin/post.go
 * @Description: https://github.com/Coder-ZhaoLu/pinkmoe 
 * 问题反馈qq群:749150798
 * xanaduCms程序上所有内容(包括但不限于 文字，图片，代码等)均为指针科技原创所有，采用请注意许可
 * 请遵循 “非商业用途” 协议。商业网站或未授权媒体不得复制内容，如需用于商业用途或者二开，请联系作者捐助任意金额即可，我们将保存所有权利。
 * Copyright (c) 2022 by 指针科技, All Rights Reserved.
 */
package admin

import (
	"server/controller"
	adminLogic "server/logic/admin"
	"server/model"
	"server/model/request"
	"server/model/response"
	"server/util"

	"github.com/gin-gonic/gin"
	"go.uber.org/zap"
)

func init() {
	controller.Register(&Post{})
}

type Post struct{}

func (post *Post) AuthPostListGet(c *gin.Context) {
	var p request.SearchPostParams
	if err := c.ShouldBindQuery(&p); err != nil {
		// 记录日志
		zap.L().Error("request.SearchPostParams with invalid param", zap.Error(err))
		response.FailWithMessage(response.CodeInvalidParam.Msg(), c)
		return
	}
	userId, _ := util.GetCurrentUserID(c)
	if err, list, total := adminLogic.GetAdminPostList(p, userId.String()); err != nil {
		zap.L().Error("adminLogic.GetPostList err", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
		return
	} else {
		response.OkWithData(response.PageResult{
			List:      list,
			PageCount: util.GetPageCount(total, p.PageSize),
			Page:      p.Page,
			PageSize:  p.PageSize,
		}, c)
	}
}

func (post *Post) AuthPostCreate(c *gin.Context) {
	var p request.CreatePostParams
	if err := c.ShouldBindJSON(&p); err != nil {
		// 记录日志
		zap.L().Error("request.CreatePostParams with invalid param", zap.Error(err))
		response.FailWithMessage(response.CodeInvalidParam.Msg(), c)
		return
	}
	if err := adminLogic.CreatePost(p); err != nil {
		zap.L().Error("adminLogic.CreatePost err", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
		return
	} else {
		response.Ok(c)
	}
}

func (post *Post) AuthPostUpdate(c *gin.Context) {
	var p request.CreatePostParams
	if err := c.ShouldBindJSON(&p); err != nil {
		// 记录日志
		zap.L().Error("request.CreatePostParams with invalid param", zap.Error(err))
		response.FailWithMessage(response.CodeInvalidParam.Msg(), c)
		return
	}
	if err := adminLogic.UpdatePost(p); err != nil {
		zap.L().Error("adminLogic.UpdatePost err", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
		return
	} else {
		response.Ok(c)
	}
}

func (post *Post) AuthPostUpdateStatus(c *gin.Context) {
	var p request.CreatePostParams
	if err := c.ShouldBindJSON(&p); err != nil {
		// 记录日志
		zap.L().Error("request.CreatePostParams with invalid param", zap.Error(err))
		response.FailWithMessage(response.CodeInvalidParam.Msg(), c)
		return
	}
	if err := adminLogic.UpdatePostStatus(p); err != nil {
		zap.L().Error("adminLogic.UpdatePostStatus err", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
		return
	} else {
		response.Ok(c)
	}
}

func (post *Post) AuthPostDelete(c *gin.Context) {
	var p model.XdPost
	if err := c.ShouldBindJSON(&p); err != nil {
		// 记录日志
		zap.L().Error("model.XdPost with invalid param", zap.Error(err))
		response.FailWithMessage(response.CodeInvalidParam.Msg(), c)
		return
	}
	if err := adminLogic.DeletePost(p); err != nil {
		zap.L().Error("adminLogic.DeletePost err", zap.Error(err))
		response.FailWithMessage(err.Error(), c)
		return
	} else {
		response.Ok(c)
	}
}
